libs/utils/utils.js

```diff
@@ -975,6 +1225,77 @@ export function convertStageLinks({ anchors, config, hostname, href }) {
   });
 }
 
+export async function decorateLinksAsync(el) {
```

Comment:
This seems like it has a rather big overlap with decorateLinks.. can we avoid some or all of the duplication? They can re-use common shared smaller functions